### PR TITLE
Issue #8598: OuterTypeFileName throws NPE on record definition in method

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8598: OuterTypeFileName throws NPE on record definition in method

Check Regression reports:
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_checks-nonjavadoc-error
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_checks-only-javadoc-error
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_part1
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_part2
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_part3
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_part4
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_part5
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_part6
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_seventu-check-regression_part_1
https://nmancus1.github.io/full-records-grammar-patch_check_diff_reports_2020_08_03/diff_seventu-check-regression_part_2

### CODE REVIEW

File: src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g [hunk 1]
```diff
@@ -1121,8 +1121,9 @@ traditionalStatement
         // up, but that's pretty hard without a symbol table ;)
         |    (declaration)=> declaration SEMI
 
-        // record declaration, note that you cannot have modifiers in this case
-        |   recordDefinition[#null]
+        // record declaration, note that you cannot have modifiers in this case so
+        // we create an empty modifiers AST to avoid NPE
```
Comment (romani) [inline src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g:1125]: `// we create an empty modifiers AST as we do for classes without modifiers`

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/outertypefilename/InputOuterTypeFilenameRecord.java [hunk 2]
```diff
@@ -1,8 +1,8 @@
 //non-compiled with javac: contains different class name by demand of test
 package com.puppycrawl.tools.checkstyle.checks.outertypefilename;
+
 /* Config:
  * default
  */
-public record IncorrectName(int x, int y, String str) { // violation
-
+public record IncorrectName1(int x, int y, String str) { // violation
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/outertypefilename/InputOuterTypeFilenameRecord.java:7]: plase move this update  in separate first "minor: ......" commit.
To explain why it is required.

File: src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g (hunk 1 above)
Comment (nmancus1) [reply src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g]: Fixed.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/outertypefilename/InputOuterTypeFilenameRecord.java (hunk 2 above)
Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/outertypefilename/InputOuterTypeFilenameRecord.java:7]: Done.

**Pull request merged**